x-pack/plugins/advanced_ui_actions/public/components/action_wizard/action_wizard.tsx

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiKeyPadMenuItem,
+} from '@elastic/eui';
+import { txtChangeButton } from './i18n';
+import './action_wizard.scss';
+
+// TODO: this interface is temporary for just moving forward with the component
+// and it will be imported from the ../ui_actions when implemented properly
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type ActionBaseConfig = {};
+export interface ActionFactory<Config extends ActionBaseConfig = ActionBaseConfig> {
+  type: string; // TODO: type should be tied to Action and ActionByType
+  displayName: string;
+  iconType?: string;
+  wizard: React.FC<ActionFactoryWizardProps<Config>>;
+  createConfig: () => Config;
+  isValid: (name: string, config: Config) => boolean;
+}
+
+export interface ActionFactoryWizardProps<Config extends ActionBaseConfig> {
+  config?: Config;
+
+  /**
+   * Callback called when user updates the config in UI.
+   */
+  onConfig: (config: Config) => void;
+}
+
+export interface ActionWizardProps {
+  /**
+   * List of available action factories
+   */
+  actionFactories: Array<ActionFactory<any>>; // any here to be able to pass array of ActionFactory<Config> with different configs
+
+  /**
+   * Currently selected action factory
+   * undefined - is allowed and means that non is selected
+   */
+  currentActionFactory?: ActionFactory;
+  /**
+   * Action factory selected changed
+   * null - means user click "change" and removed action factory selection
+   */
+  onActionFactoryChange: (actionFactory: ActionFactory | null) => void;
+
+  /**
+   * current config for currently selected action factory
+   */
+  config?: ActionBaseConfig;
+
+  /**
+   * config changed
+   */
+  onConfigChange: (config: ActionBaseConfig) => void;
+}
+export const ActionWizard: React.FC<ActionWizardProps> = ({
```

Comment:
Complete nit: every time I pull down this code I add new lines above the exported stuff to make it easier to read/scan.